Documentation/MyFirstContribution

```diff
@@ -0,0 +1,584 @@
+button. Place your fork wherever you deem appropriate and create it.
+
+=== Uploading To Your Own Fork
+
+To upload your branch to your own fork, you'll need to add the new fork as a
+remote. You can use `git remote -v` to show the remotes you have added already.
+From your new fork's page on GitHub, you can press "Clone or download" to get
+the URL; then you need to run the following to add, replacing your own URL and
+remote name for the examples provided:
+
+----
+git remote add remotename git@github.com:remotename/git.git
+----
+
+Run `git remote -v` again and you should see the new remote showing up.
+`git fetch remotename` (with the real name of your remote replaced) in order to
+get ready to push.
+
+Next, double-check that you've been doing all your development in a new branch
+by running `git branch`. If you didn't, now is a good time to move your new
+commits to their own branch.
+
+As mentioned briefly at the beginning of this doc, we are basing our work on
+master, so go ahead and update as shown below, or using your preferred
+workflow.
+
+----
+git checkout master
+git pull -r
+git rebase master psuh
+----
+
+Finally, you're ready to push your new topic branch! (Due to our branch and
+command name choices, be careful when you type the command below.)
+
+----
+git push remotename psuh
+----
+
+Now you should be able to go and check out your newly created branch on GitHub.
+
+////
+TODO: The next few bullets describe testing and pushing your change with
+GitGitGadget. It may be useful to describe a workflow using git send-email as
+well.
+////
+
+=== Sending a PR to GitGitGadget
+
+GitGitGadget is a tool created by Johannes Schindelin to make life as a Git
+contributor easier for those used to the GitHub PR workflow. It's documented
+at gitgitgadget.github.io. In order to have your code tested and formatted for
+review, you need to start by opening a Pull Request against gitgitgadget/git.
+Head to https://github.com/gitgitgadget/git and open a PR either with the "New
+pull request" button or the convenient "Compare & pull request" button that may
+appear with the name of your newly pushed branch.
+
+Review the PR's title and description, as it's used by GitGitGadget as the cover
+letter for your change. When you're happy, submit your pull request.
+
+=== Getting CI to Run
+
+If it's your first time using GitGitGadget (which is likely, as you're using
+this tutorial) then someone will need to give you permission to use the tool.
+As mentioned in the GitGitGadget doc, you just need someone who already uses it
+to comment on your PR with `/allow <username>`.
+
+Once you're whitelisted, GitGitGadget will automatically run your PRs through
+the CI.
+
+If the CI fails, you can update your changes with `rebase -i` and push your
+branch again:
+
+----
+git push -f remotename psuh
+----
+
+In fact, you should continue to make changes this way up until the point when
+your patch is accepted into `next`.
```

Comment:
This `next` comes a bit out of left field.

How about explaining first that there are four integration branches in Git (`pu`, `next`, `master`, `maint`), and that new contributions are usually kept in `pu` until they are deemed ready for early testers who are encouraged to run `next`?